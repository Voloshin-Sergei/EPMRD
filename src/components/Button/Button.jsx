import React from 'react';

import style from './Button.module.scss';

export const Button = ({ text, className, filled }) => (
  <button className={`${style.button} ${className} ${filled ? style.filled : ''}`} type="button">
    {text}
  </button>
);
