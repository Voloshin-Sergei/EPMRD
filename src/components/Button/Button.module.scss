@import 'Styles/variables/colors';

.button {
  border: none;
  cursor: pointer;
}

.filled {
  padding: 5px 10px;
  font-size: 1.2rem;
  text-transform: uppercase;
  color: $color-primary;
}
