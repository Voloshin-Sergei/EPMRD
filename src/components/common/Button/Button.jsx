import React from 'react';
import propTypes from 'prop-types';
import style from './Button.module.scss';

export const Button = ({
  className, variant = 'primary', text, type = 'button',
}) => (
  <button className={`${style.button} ${className} ${style[variant]}`} type={type}>
    {text}
  </button>
);

Button.propTypes = {
  className: propTypes.string.isRequired,
  variant: propTypes.oneOf(['primary', 'secondary']).isRequired,
  text: propTypes.string.isRequired,
  type: propTypes.oneOf(['button', 'submit']).isRequired,
};
