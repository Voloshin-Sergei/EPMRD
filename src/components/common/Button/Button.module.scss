@import 'Styles/variables/colors';

.button {
  border: none;
  cursor: pointer;
}

.primary {
  padding: 0.5rem 1rem;
  font-size: 1.2rem;
  text-transform: uppercase;
  color: $color-primary;
}

.secondary {
  background-color: transparent;
}