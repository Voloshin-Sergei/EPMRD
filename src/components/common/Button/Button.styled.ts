import styled, { css } from 'styled-components';
import { colors } from '../../../styles/variables/colors.styled';

interface ButtonStyledProps {
  variant: 'primary' | 'secondary';
  typeBtn: 'searchBtn' | 'sortTagBtn' | 'filterTagBtn' | undefined;
  active: boolean | undefined;
}

const primaryStyle = css`
  padding: 0.5rem 1rem;
  color: ${colors.colorPrimary};
  font-size: 1.2rem;
  text-transform: uppercase;
`;

const sortTagBtn = css`
  background-color: ${colors.colorPrimaryDarkLighten};

  :hover {
    background-color: ${colors.colorPrimaryDarken};
  }
`;

const sortTagBtnActive = css`
  background-color: ${colors.colorSecondary};

  :hover {
    background-color: ${colors.colorSecondary};
  }
`;

const filterTagBtn = css`
  font-size: 1.6rem;
  transform: scale(0.9);

  :hover {
    transform: scale(1);
  }
`;

const filterTagBtnActive = css`
  color: ${colors.colorSecondary};
  font-size: 1.8rem;

  :hover {
    transform: scale(0.9);
  }
`;

const searchBtn = css`
  margin-left: auto;
  font-size: 1.8rem;
  background-color: ${colors.colorSecondary};

  :hover {
    background-color: ${colors.colorSecondaryLighten};
  }

  :active {
    transform: scale(0.9);
  }
`;

const secondaryStyle = css`
  background-color: transparent;
`;

export const ButtonStyled = styled.button<ButtonStyledProps>`
  border: none;
  cursor: pointer;

  ${(props) => (props.variant === 'primary' ? primaryStyle : secondaryStyle)};
  ${(props) => (props.typeBtn === 'searchBtn' ? searchBtn : '')};
  ${(props) => (props.typeBtn === 'sortTagBtn' ? sortTagBtn : '')};
  ${(props) => (props.typeBtn === 'sortTagBtn' && props.active ? sortTagBtnActive : '')};
  ${(props) => (props.typeBtn === 'filterTagBtn' ? filterTagBtn : '')};
  ${(props) => (props.typeBtn === 'filterTagBtn' && props.active ? filterTagBtnActive : '')};
`;
