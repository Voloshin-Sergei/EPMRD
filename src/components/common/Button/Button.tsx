import React from 'react';
import style from './Button.module.scss';

export interface ButtonProps {
  className?: string,
  variant?: 'primary' | 'secondary',
  children: React.ReactNode,
  type?: 'button' | 'submit'
}

export const Button: React.FC<ButtonProps> = ({
  className='', variant = 'primary', children, type = 'button',
}) => (
  <button className={`${style.button} ${className} ${style[variant]}`} type={type}>
    {children}
  </button>
);
