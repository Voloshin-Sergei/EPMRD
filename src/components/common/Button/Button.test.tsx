import React from 'react';
import { shallow } from 'enzyme';
import { Button } from './Button';

describe("Button component", () => {
  it('should render Button component', () => {
    const component = shallow(<Button>{'test button'}</Button>);
    expect(component).toMatchSnapshot();
  });

  it('should render Button component with className props', () => {
    const component = shallow(<Button className="test_class">{'test button'}</Button>);
    expect(component).toMatchSnapshot();
  });

  it('should render Button component with variant props', () => {
    const component = shallow(<Button variant="secondary">{'test button'}</Button>);
    expect(component).toMatchSnapshot();
  });

  it('should render Button component with type props', () => {
    const component = shallow(<Button type="submit">{'test button'}</Button>);
    expect(component).toMatchSnapshot();
  });
});