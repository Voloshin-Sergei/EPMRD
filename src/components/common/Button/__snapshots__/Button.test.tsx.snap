// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component should render Button component 1`] = `
<button
  className="button  primary"
  type="button"
>
  test button
</button>
`;

exports[`Button component should render Button component with className props 1`] = `
<button
  className="button test_class primary"
  type="button"
>
  test button
</button>
`;

exports[`Button component should render Button component with type props 1`] = `
<button
  className="button  primary"
  type="submit"
>
  test button
</button>
`;

exports[`Button component should render Button component with variant props 1`] = `
<button
  className="button  secondary"
  type="button"
>
  test button
</button>
`;
